tool/scaffold/src/main/java/org/openscience/cdk/tools/scaffold/ScaffoldGenerator.java

```diff
@@ -0,0 +1,2412 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package org.openscience.cdk.tools.scaffold;
+
+import org.openscience.cdk.CDKConstants;
+import org.openscience.cdk.aromaticity.Aromaticity;
+import org.openscience.cdk.aromaticity.ElectronDonation;
+import org.openscience.cdk.exception.CDKException;
+import org.openscience.cdk.fragment.MurckoFragmenter;
+import org.openscience.cdk.graph.ConnectivityChecker;
+import org.openscience.cdk.graph.CycleFinder;
+import org.openscience.cdk.graph.Cycles;
+import org.openscience.cdk.interfaces.IAtom;
+import org.openscience.cdk.interfaces.IAtomContainer;
+import org.openscience.cdk.interfaces.IAtomContainerSet;
+import org.openscience.cdk.interfaces.IAtomType;
+import org.openscience.cdk.interfaces.IBond;
+import org.openscience.cdk.interfaces.IChemObject;
+import org.openscience.cdk.interfaces.IRingSet;
+import org.openscience.cdk.interfaces.IStereoElement;
+import org.openscience.cdk.smiles.SmiFlavor;
+import org.openscience.cdk.smiles.SmilesGenerator;
+import org.openscience.cdk.tools.CDKHydrogenAdder;
+import org.openscience.cdk.tools.ILoggingTool;
+import org.openscience.cdk.tools.LoggingToolFactory;
+import org.openscience.cdk.tools.manipulator.AtomContainerManipulator;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+import java.util.TreeMap;
+
+/**
+ * This class is designed to generate different molecule scaffolds and frameworks.
+ * It contains several other methods for the decomposition of molecules.
+ * All details of the functionality and implementation are described <a href="https://doi.org/10.26434/chemrxiv-2022-7tf0h-v2">
+ * here</a>.<p>
+ *
+ * Furthermore, the molecules can be decomposed according to the <a href="https://doi.org/10.1021/ci600338x">
+ * Schuffenhauer rules</a> and a generation of all possible fragments,
+ * which can be created by the iterative removal of the rings, is also possible. <br>
+ * The resulting molecular fragments can be organised in the form of a tree or a network.
+ * The network approach is based on the <a href="https://doi.org/10.1021/ci2000924">
+ * Mining for Bioactive Scaffolds with Scaffold Networks</a> paper.
+ * Different trees or networks can also be merged together.
+ * The Scaffold Generator library is described in <a href="https://doi.org/10.1186/s13321-022-00656-x"> Schaub et al.
+ * "Scaffold Generator: a Java library implementing molecular scaffold functionalities in the Chemistry Development Kit
+ * (CDK)" (J Cheminform 14, 79, 2022)</a>
+ *
+ * @author Julian Zander, Jonas Schaub (zanderjulian@gmx.de, jonas.schaub@uni-jena.de)
+ * @version 1.0.4.0
+ */
+public class ScaffoldGenerator {
+
+    //<editor-fold desc="Enum">
```

Comment:
Plz check all code. See above.

Reply:
Editor folds were removed everywhere with the latest commit.